### PR TITLE
Add event support

### PR DESCRIPTION
**Pull request by YoungHS-NUS**

### CODE REVIEW

File: solc_json_parser/parser.py [hunk 1]
```diff
@@ -302,8 +305,8 @@ def _get_modifiers(node: Dict) -> List[str]:
         func_kind = node.get('kind')
         state_mutability = node.get('stateMutability')
 
-        signature = _get_signature(name, parameters, func_kind)
-        return_signature = _get_signature("", return_type, func_kind)
+        signature = self.get_signature(name, parameters, func_kind)
```
Comment (cl2089) [inline solc_json_parser/parser.py:308]: `func_kind` implies different kind of function, maybe we could simply name it as `kind`?

File: solc_json_parser/parser.py [hunk 2]
```diff
@@ -411,7 +435,7 @@ def _add_inherited_function_fields(data_dict: Dict[int, ContractData]):
                             new_function.inherited_from = base_contract_name
                             contract.functions.append(new_function)
 
-        # self.save_solc_ast_json("Storage_solc_ast_no_abstractBugC")
+        # self.save_solc_ast_json("event")
```
Comment (cl2089) [inline solc_json_parser/parser.py:438]: please remove unused code if no longer useful

File: tests/test_parser.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import unittest
 from solc_json_parser.parser import SolidityAst, SolidityAstError
 contracts_root = './contracts'
+from pprint import pprint
```
Comment (cl2089) [inline tests/test_parser.py:4]: not used, please remove

File: tests/test_parser.py [hunk 4]
```diff
@@ -303,3 +303,54 @@ def test_add_automatic_retrying(self):
             ast = SolidityAst(f'{contracts_root}/dev/buggy_10.sol', retry_num=0)
         except SolidityAstError:
             print("SolidityAstError is expected")
+
+
+    def test_event(self):
+        def sub_test(ast):
+            events = ast.events_in_contract_by_name('IPoolEvents')
+            self.assertEqual(15, len(events))
```
Comment (cl2089) [inline tests/test_parser.py:311]: To make the test more robust and readable, I suggest  creating expected event list and comparing it with the actual events list, for example replacing the code in `sub_test` with
```
assertEqual(set(expected_events_excluding_raw), set(actual_events_excluding_raw))
```

File: solc_json_parser/parser.py (hunk 2 above)
Comment (YoungHS-NUS) [reply solc_json_parser/parser.py:438]: This is useful when developing new features or debugging.

**Comment by cl2089** [comment]
hi @YoungHS-NUS, most comments are informational except [this one](https://github.com/sbip-sg/solc-json-parser/pull/37#discussion_r1047950078)

**Comment by YoungHS-NUS** [comment]
Will PR tomorrow, including the change to handle importing from ../

Handling ../ also varies from versions. I will write a more detailed readme later

**Pull request closed without merging**